code/espurna/wifi.ino

```diff
@@ -453,15 +453,16 @@ void _wifiWebSocketOnConnected(JsonObject& root) {
     root["max"] = WIFI_MAX_NETWORKS;
     wifi["scan"] = getSetting("wifiScan", 1 == WIFI_SCAN_NETWORKS);
 
-    const char* keys[] = {
-        "ssid", "pass", "ip", "gw", "mask", "dns", "stored"
-    };
-  
-    // TODO check schema
     JsonArray& schema = wifi.createNestedArray("schema");
-    schema.copyFrom(keys, 7);
 
-    JsonArray& networks = wifi.createNestedArray("networks");
+    schema.add("ssid");
+    schema.add("pass");
+    schema.add("ip");
+    schema.add("gw");
+    schema.add("mask");
+    schema.add("dns");
```

Comment:
Note that I was kind of doing the same thing, hardcoded 7 can also be (sizeof(keys)/sizeof(keys[0]))
I changed the behaviour of wifi settings, getSetting() will return hardcoded default. `stored` flag is to avoid removing entry accidentally, since on ui reload it will appear again anyway

Reply:
I just didn't want to use copyFrom because it was removed from version 6 of the Json array lib (Using v5 but if ever we move away)
Also not sure but doesn't it double the ram usage for the array (define + clone, seems redundant)

I don't get the stored flag
Don't `if (!getSetting({"ssid", index}, _wifiSSID(index)).length()) break;` prevent it from ever being set to false anyway?

Reply:
See https://arduinojson.org/v6/doc/upgrade/#copyarray

When we have `WIFI1_SSID=something`, getsetting is never empty

Reply:
Oh I finally got it, so the WIFI can be hardcoded with defines and this flag tells us if it comes from the define or not

But can't they be overwritten? The defines acting just as a default value

Reply:
Yes, see `code/espurna/wifi_config.h`
They are hard-coded defaults, we can't erase them. It is still possible to "shadow" them, either by setting indexed setting to something else or setting "" empty value

Reply:
Okay so I got it all but then there might be an issue, if you overwrite the settings the delete button will still be disabled, but it shouldn't (it wouldn't technically delete it, but would reset it to default)
So maybe just changing the name of the button to 'Reset to default' instead of 'Delete'

Having a hardcoded flag sometimes complicates things more than it simpifies it :p

Reply:
Kind of like that. You can always hit reset-all button though, I guess there could be a specific 'reset wifi settings' one.
pre 1.14.2 versions read hardcoded settings on boot, overwriting runtime settings with those values, so it kind of had the same problem only delayed until the next reset

Reply:
btw, not to speculate on code improvements regarding arduinojson:
https://github.com/xoseperez/espurna/pull/1957#issuecomment-545905580

you can use it as a local library and build program on host. you can repeat the code from above for yourself, array size does not change when we copyfrom, it only stores pointers since it is array of pointers

Reply:
Yeah that's what I figured, so there is no difference between the two? It's a matter of personal preference then

Reply:
True, effectively it would be the exact same code generated for v5:
https://github.com/bblanchon/ArduinoJson/blob/v5.13.4/src/ArduinoJson/JsonArray.hpp#L151
Slightly different for v6
https://github.com/bblanchon/ArduinoJson/blob/8f8c82d400a2e860e538eed8f92674330186f76f/src/ArduinoJson/Array/Utilities.hpp#L47
But [index] access is also available in v5 (not sure what empty values are though, nulls?)